lib/puppet/type/zfs_share.rb

```diff
@@ -0,0 +1,234 @@
+Puppet::Type.newtype(:zfs_share) do
+  @doc = "Manage zfs shares"
+
+  require 'puppet/parameter/boolean'
+
+  ensurable
+
+  # zfs share documentation
+  # https://docs.oracle.com/cd/E26502_01/html/E29031/zfs-share-1m.html
+
+  newparam(:name) do
```

Comment:
The more I think it through the more I like the idea of using name as the "named share" parameter.  The name of the share that is generated by the system is predictable.  It means changing the "share_title" parameter to something like "filesystem" and if "filesystem" and "name" do not match then set the share's zfs share property name to the "name" parameter and path to "filesystem" parameter.  This just seems to produce a more intuitive naming scheme.